src/v/ssx/future-util.h

```diff
@@ -268,12 +268,14 @@ inline constexpr detail::background_t background;
 /// \brief Create a new future, handling common shutdown exception types.
 inline seastar::future<>
 ignore_shutdown_exceptions(seastar::future<> fut) noexcept {
-    return std::move(fut)
-      .handle_exception_type([](const seastar::abort_requested_exception&) {})
-      .handle_exception_type([](const seastar::gate_closed_exception&) {})
-      .handle_exception_type([](const seastar::broken_semaphore&) {})
-      .handle_exception_type([](const seastar::broken_promise&) {})
-      .handle_exception_type([](const seastar::broken_condition_variable&) {});
+    try {
+        co_await std::move(fut);
```

Comment:
This seems good since `handle_exception_type` is relatively expensive whether an exception occurs or not, but we this transformation means we only take the exception matching cost when an exception is actually thrown.

Reply:
I guess that was the intent of this transformation?

Reply:
Based on my reading of https://github.com/redpanda-data/seastar/blob/245e0ccfa6d58d7e0dca2b4034ce1bc43e39bdc5/include/seastar/core/future.hh#L1858-L1862

It seems like before we had 5 continuations and 5 try catch blocks, and now we only have 1 of each (I think, I haven't fully wrapped my brain around how a compiler transforms coroutines). That was the intent here to cut down on both of those in a hope to lessen the cost of this function you noticed in the profiles.